hpx/lcos/local/condition_variable.hpp

```diff
@@ -194,6 +205,10 @@ namespace hpx { namespace lcos { namespace local
             util::ignore_all_while_checking ignore_lock;
             std::unique_lock<mutex_type> l(mtx_);
             util::unlock_guard<Lock> unlock(lock);
+            //The following ensures that the inner lock will be unlocked
+            //before the outer to avoid deadlock (fixes issue #3608)
+            std::lock_guard <std::unique_lock<mutex_type> > unlock_next
+            (l, std::adopt_lock);
```

Comment:
Please use clang-format to adapt the formatting of the newly added lines to our coding guidelines.